discovery-frontend/src/app/data-preparation/dataflow/dataflow-detail/component/edit-dataflow-rule/rule-union-popup/rule-union-popup.component.html

```diff
@@ -54,7 +54,13 @@
             <ul class="ddp-list-union" *ngFor="let colInfo of resultFields; let i = index">
               <li [ngClass]="{'ddp-selected':editResultColumnName == colInfo.name}">
                 <div class="ddp-box-column" >
-                  <em class="ddp-icon-type-ab"></em>
+                  <em class="ddp-icon-type-ab"  *ngIf="colInfo.type === 'STRING' || colInfo.type === 'USER_DEFINED' || colInfo.type === 'TEXT'"></em>
+                  <em class="ddp-icon-type-local" *ngIf="colInfo.type === 'LNG' || colInfo.type === 'LNT'"></em>
+                  <em class="ddp-icon-type-calen" *ngIf="colInfo.type === 'TIMESTAMP'"></em>
+                  <em class="ddp-icon-type-sharp" *ngIf="colInfo.type === 'LONG' || colInfo.type === 'INTEGER' || colInfo.type === 'DOUBLE' || colInfo.type === 'CALCULATED'"></em>
+                  <em class="ddp-icon-type-tf" *ngIf="colInfo.type === 'BOOLEAN'"></em>
+                  <em class="ddp-icon-type-array" *ngIf="colInfo.type === 'ARRAY'"></em>
+                  <em class="ddp-icon-type-map" *ngIf="colInfo.type === 'MAP'"></em>
```

Comment:
I think it's duplicate code.;; could you replace this using some kind of type-check function. 
Just my opinion, please refer to it :)

Reply:
that is applied